observer/bin/spark-observer.js

```diff
@@ -3,27 +3,50 @@ import { ethers } from 'ethers'
 import * as Sentry from '@sentry/node'
 import timers from 'node:timers/promises'
 
-import { RPC_URL, rpcHeaders, OBSERVATION_INTERVAL_MS } from '../lib/config.js'
-import { getPgPool } from '../lib/db.js'
-import { observe } from '../lib/observer.js'
+import { RPC_URL, rpcHeaders } from '../lib/config.js'
+import { getPgPools } from '@filecoin-station/spark-stats-db'
+import {
+  observeTransferEvents,
+  observeScheduledRewards
+} from '../lib/observer.js'
 
-const pgPool = await getPgPool()
+const pgPools = await getPgPools()
 
 const fetchRequest = new ethers.FetchRequest(RPC_URL)
 fetchRequest.setHeader('Authorization', rpcHeaders.Authorization || '')
 const provider = new ethers.JsonRpcProvider(fetchRequest, null, { polling: true })
 
 const ieContract = new ethers.Contract(SparkImpactEvaluator.ADDRESS, SparkImpactEvaluator.ABI, provider)
 
-while (true) {
-  const start = new Date()
-  try {
-    await observe(pgPool, ieContract, provider)
-  } catch (e) {
-    console.error(e)
-    Sentry.captureException(e)
-  }
-  const dt = new Date() - start
-  console.log(`Observation took ${dt}ms`)
-  await timers.setTimeout(OBSERVATION_INTERVAL_MS - dt)
-}
+const ONE_HOUR = 60 * 60 * 1000
+
+await Promise.all([
+  (async () => {
+    while (true) {
+      const start = new Date()
+      try {
+        await observeTransferEvents(pgPools, ieContract, provider)
+      } catch (e) {
+        console.error(e)
+        Sentry.captureException(e)
+      }
+      const dt = new Date() - start
+      console.log(`Observing Transfer events took ${dt}ms`)
+      await timers.setTimeout(ONE_HOUR - dt)
+    }
+  })(),
+  (async () => {
+    while (true) {
+      const start = new Date()
+      try {
+        await observeScheduledRewards(pgPools, ieContract, provider)
+      } catch (e) {
+        console.error(e)
+        Sentry.captureException(e)
+      }
+      const dt = new Date() - start
+      console.log(`Observing scheduled rewards took ${dt}ms`)
+      await timers.setTimeout((24 * ONE_HOUR) - dt)
+    }
+  })()
+])
```

Comment:
Would you mind extracting the inner loop functions into standalone functions? I am not a big fan of long nested lambdas.

```js
await Promise.all([
  loopObserveTransferEvents(),
  loopObserveScheduledRewards(),
])
```